### PR TITLE
VideoCLIP training script

### PR DESCRIPTION
**Pull request by sophiazhi**
Summary:
Training script for VideoCLIP on MUGEN dataset.

- `definitions.py`: add `TrainingArgs`, which includes lightning Trainer args like `max_epochs`
- `train.py`: training script
- `configs/train.yaml`: training config 
- `configs/eval.yaml`: fix default resolution value  ([default dataset args](https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/data/mugen_data.py#L52), [eval command](https://github.com/mugen-org/MUGEN_baseline/blob/b45da493f3ddbcced52d44eee02d443f90842474/retrieval/experiments/eval_all.sh))
- `model.py`: remove recall calculation at end of training epoch, add split markers to recall logging (e.g. test/recall and validation/recall)

Test plan:
`python train.py config=configs/train.yaml`
[tensorboard](https://colab.research.google.com/drive/1RfZBYqpxmGiNCna-70_xm2ce85eEjB9I?usp=sharing) of logged results

### CODE REVIEW

File: examples/mugen/retrieval/model.py [hunk 1]
```diff
@@ -106,15 +106,6 @@ def training_step(self, batch, batch_idx):
         )
         return {"loss": loss, "model_output": model_output}
 
-    def training_epoch_end(self, outputs):
```
Comment (langong347) [inline examples/mugen/retrieval/model.py:109]: Just to confirm: this is why previously recall was computed at every epoch?

File: examples/mugen/retrieval/train.py [hunk 2]
```diff
@@ -47,7 +47,8 @@ def train():
 
     trainer = Trainer(
         accelerator=args.accelerator,
-        devices=1,
+        devices=args.devices,
+        strategy="ddp_find_unused_parameters_false",
```
Comment (langong347) [inline examples/mugen/retrieval/train.py:51]: How do we know there's no unused parameter?

**Comment by langong347** [comment]
>fix default resolution value

The hyperlink there led to command line `python retrieval/eval.py --get_game_frame --get_audio --get_text_desc --use_manual_annotation`. Should we link to the dataset args instead?

https://github.com/mugen-org/MUGEN_baseline/blob/main/lib/data/mugen_data.py#L52

**Review by langong347 — APPROVED** [review]

File: examples/mugen/retrieval/model.py (hunk 1 above)
Comment (sophiazhi) [reply examples/mugen/retrieval/model.py:109]: correct, that function is why recall was computed for the entire training set at every epoch (recall is still computed at every epoch for the validation set in `validation_epoch_end`)

File: examples/mugen/retrieval/train.py (hunk 2 above)
Comment (sophiazhi) [reply examples/mugen/retrieval/train.py:53]: when i didn't have this argument, the script printed a warning saying no unused params were found and having `find_unused_parameters=True` makes performance worse

File: examples/mugen/retrieval/train.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    dataset_args: MUGENDatasetArgs = args.dataset_args
+    datamodule = MUGENDataModule(
+        dataset_args,
+        text_transform=BertTextTransform(
+            **vars(args.datamodule_args.bert_text_transform)
+        ),
+        video_transform=VideoTransform(**vars(args.datamodule_args.video_transform)),
+        batch_size=args.datamodule_args.batch_size,
+        num_workers=args.datamodule_args.num_workers,
+        shuffle=args.datamodule_args.shuffle,
```
Comment (langong347) [inline examples/mugen/retrieval/train.py:41]: Does this evaluate to `False` as the default in the yaml? In MUGEN `shuffle` is `True` during training.

Comment (langong347) [reply examples/mugen/retrieval/train.py:41]: https://github.com/mugen-org/MUGEN_baseline/blob/b45da493f3ddbcced52d44eee02d443f90842474/retrieval/train.py#L21
